pkg/target/target.go

```diff
@@ -141,18 +141,40 @@ func (m *Manager) BundlesForCluster(cluster *fleet.Cluster) (result []*fleet.Bun
 
 		cgs, err := m.clusterGroupsForCluster(cluster)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
-		m := bundle.Match(cluster.Name, ClusterGroupsToLabelMap(cgs), cluster.Labels)
-		if m != nil {
+		match := bundle.Match(cluster.Name, ClusterGroupsToLabelMap(cgs), cluster.Labels)
+		if match != nil {
 			result = append(result, app)
+		} else {
+			bd, err := m.getBundleDeploymentsInCluster(app, cluster)
+			if err != nil {
+				return nil, nil, err
+			}
+			bdCleanupSet = append(bdCleanupSet, bd...)
 		}
 	}
 
 	return
 }
 
+func (m *Manager) getBundleDeploymentsInCluster(app *fleet.Bundle, cluster *fleet.Cluster) (result []*fleet.BundleDeployment, err error) {
+	bundleDeployments, err := m.bundleDeploymentCache.List("", labels.SelectorFromSet(DeploymentLabelsForSelector(app)))
+	if err != nil {
+		return nil, err
+	}
+	str := []string{"cluster", cluster.Namespace, cluster.Name}
+	nsPrefix := strings.Join(str, "-")
+	for _, appDep := range bundleDeployments {
+		if strings.HasPrefix(appDep.Namespace, nsPrefix) {
+			result = append(result, appDep)
+		}
+	}
```

Comment:
nit: `appDep` sounds weird, can we rename this `bd`?

Reply:
done